### ISSUE TITLE
Several issues with Swagger-UI v2.1.0-M2 and a Swagger v1.2 doc

### ISSUE THREAD

**Issue by xiehan** [swagger-api/swagger-ui#1147]
N.B. Upgrading to Swagger v2.0 isn't an option because Swagger-PHP doesn't support it yet (at least not stable enough to make it feasible for us).

I apologize for the catch-all/dump of images but I'm heading out of town in less than half an hour and wanted to make sure I report these issues while they're fresh on my mind / before I forget.

---

**Problems rendering nested models:**

Before upgrading to v2.1.0-M2:

![screen shot 2015-04-10 at 4 28 00 pm](https://cloud.githubusercontent.com/assets/855115/7096818/9a859fe0-df9e-11e4-8b33-e9caac35b802.png)

After:

![screen shot 2015-04-10 at 4 24 47 pm](https://cloud.githubusercontent.com/assets/855115/7096821/9f82b5f0-df9e-11e4-9555-9da63569f285.png)

---

**Problems rendering enums (as dropdowns):**

Before upgrading to v2.1.0-M2:

![screen shot 2015-04-10 at 4 29 06 pm](https://cloud.githubusercontent.com/assets/855115/7096846/d3011746-df9e-11e4-9ddf-872dc35e724e.png)

After:

![screen shot 2015-04-10 at 4 29 25 pm](https://cloud.githubusercontent.com/assets/855115/7096848/d7aa34f8-df9e-11e4-8cd0-7d559b4137df.png)

---

**Problems rendering arrays of JSON objects in the `body` field:**

Before upgrading to v2.1.0-M2:

![screen shot 2015-04-10 at 4 31 17 pm](https://cloud.githubusercontent.com/assets/855115/7096898/2ce85dd2-df9f-11e4-8bf2-927d5b5e708d.png)

After:

![screen shot 2015-04-10 at 4 31 30 pm](https://cloud.githubusercontent.com/assets/855115/7096899/31be8bec-df9f-11e4-90f0-35f340f14a35.png)

---

**Error response models are just completely missing (and I'm not sure what's supposed to be going on with the `default` vs. `201`):**

Before upgrading to v2.1.0-M2:

![screen shot 2015-04-10 at 4 35 38 pm](https://cloud.githubusercontent.com/assets/855115/7096977/bcdc5358-df9f-11e4-8d51-4c121f00d9ca.png)

After:

![screen shot 2015-04-10 at 4 35 25 pm](https://cloud.githubusercontent.com/assets/855115/7096980/c2cf83c0-df9f-11e4-8a8b-4684a8c2bb39.png)

---

**Empty response class appeared for an operation that doesn't have a 200 or 201 response:**

Before upgrading to v2.1.0-M2:

![screen shot 2015-04-10 at 4 42 45 pm](https://cloud.githubusercontent.com/assets/855115/7097123/c70ef384-dfa0-11e4-9049-5e2d89e7b19d.png)

After:

![screen shot 2015-04-10 at 4 44 05 pm](https://cloud.githubusercontent.com/assets/855115/7097147/e0c64692-dfa0-11e4-9ef9-041f03bd0747.png)

---

**Finally, this last one might be partly our fault, but the way Swagger-PHP uses annotated comments makes it really hard to avoid this one. Markdown now parsing extra spaces and interpreting text as a code block:**

Before upgrading to v2.1.0-M2:

![screen shot 2015-04-10 at 4 39 19 pm](https://cloud.githubusercontent.com/assets/855115/7097167/16c8ee66-dfa1-11e4-80c2-52d64ded298f.png)

After:

![screen shot 2015-04-10 at 4 39 38 pm](https://cloud.githubusercontent.com/assets/855115/7097169/1ada5616-dfa1-11e4-9196-01ba35a9493d.png)

JSON that's powering this:

```json
{
    "notes": "Please be aware that the required parameters are contingent on the `grant_type` that you select.\n\n     For the `authorization_code` grant type, you are **required** to pass in the `code` and `redirect_uri` parameters. `service`, `username` and `password` parameters will be ignored.\n\n     For the `client_credentials` grant type, you do not need to pass in any additional parameters beyond the basic requirements. `code`, `redirect_uri`, `service`, `username` and `password` parameters will be ignored.\n\n     For the `device_code` grant type, you are **required** to pass in the `code` parameter. `redirect_uri`, `service`, `username` and `password` parameters will be ignored.\n\n     For the `password` grant type, you are **required** to pass in the `username` and `password` parameters. If you are logging in via a social service, you should also pass in the `service` parameter. (In this case, the access token you receive from the social service will serve as both your username and password.) The `code` and `redirect_uri` parameters are ignored.\n     Third-party developers do not have access to this grant type.\n\n     The `temporary_user` grant type is a custom grant type created by NPR to suit our needs for functionality such as our &quot;try-before-you-buy&quot; experience. If you are a third-party developer, you will not have access to this grant type unless we have explicitly given you permission within our system.\n     For this grant type, you do not need to pass in any additional parameters beyond the basic requirements. `code`, `redirect_uri`, `service`, `username` and `password` parameters will be ignored.\n\n     If you are unsure of which grant type to select, assume that `authorization_code` is the one you want."
}
```

**Comment by fehguy** [comment]
Thanks for such a detailed report.  Looking into it.

**Comment by fehguy** [comment]
@xiehan ^^

**Comment by webron** [comment]
@xiehan - several of these issues have been addressed in the develop_2.0 branch. Can you please retest and verify which still exist?

Also, as @fehguy mentioned, it would help to get the JSON you use to further debug the other issues.

**Comment by llvasconcellos** [comment]
I've just tested develop_2.0 branch using the default petshop json and here is the problem that is worring me:

![capturar](https://cloud.githubusercontent.com/assets/1710823/7494381/a77833ae-f3dd-11e4-9646-ffe8a160bc5c.PNG)

**Comment by fehguy** [comment]
@llvasconcellos this is fixed, grab the latest from develop_2.0.  @xiehan if you can retest and give some more info, i'd be happy to dig in more.

**Comment by xiehan** [comment]
@fehguy Thanks for looking into these! Sorry for the delay; we've shifted the focus away from our API documentation for a few weeks to work on other things, so I haven't had a chance to check out the changes. I should have time to retest on Monday and let you guys know if things are looking better for us.

We did finally make our Swagger docs publicly accessible so if you want to try it with our data, it's at http://api.npr.org/documentation/beryllium/api-docs

As a side note, I have 2 full days next week when I can volunteer my time to help out with open source projects. If I encounter more bugs after my retesting on Monday, I may be able to address them myself, but also let me know if there are other open issues I can help out with.
